morpheus-graphql-client/test/Case/JSON/Custom/Errors.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+module Case.JSON.Custom.Errors
+  ( test,
+  )
+where
+
+import Data.ByteString.Lazy.Char8
+  ( ByteString,
+  )
+import Data.Morpheus.Types.Error
+  ( GQLError (..),
+    Position (..)
+  )
+import Data.Morpheus.Client
+  ( EncodeScalar (..),
+    Fetch (..),
+    FetchError (..),
+    ScalarValue (..),
+    gql,
+  )
+import Data.List.NonEmpty
+  ( NonEmpty(..),
+  )
+import Data.Text (Text)
+import Spec.Utils
+  ( defineClientWithJSON,
+    mockApi,
+  )
+import Test.Tasty
+  ( TestTree,
+  )
+import Test.Tasty.HUnit
+  ( assertEqual,
+    testCase,
+  )
+import Prelude
+  ( Either (..),
+    Eq (..),
+    IO,
+    Maybe (..),
+    Show,
+    ($),
+  )
+
+newtype GitTimestamp = GitTimestamp
+  { unGitTimestamp :: Text
+  }
+  deriving (Eq, Show)
+
+instance EncodeScalar GitTimestamp where
+  encodeScalar (GitTimestamp x) = String x
+
+defineClientWithJSON
+  "JSON/Custom"
+  [gql|
+    query TestQuery
+      {
+        queryTypeName
+      }
+  |]
+
+resolver :: ByteString -> IO ByteString
+resolver = mockApi "JSON/Custom/Errors"
+
+client :: IO (Either (FetchError TestQuery) TestQuery)
+client = fetch resolver ()
+
+test :: TestTree
+test = testCase "test Errors" $ do
+  value <- client
+  assertEqual
+    "test custom Errors"
+    ( Left
+        ( FetchErrorProducedErrors
+            ((GQLError "Failure" [Position {line = 3, column = 7}] Nothing Nothing) :| [])
```

Comment:
The biggest point of uncertainty was here where in master the `GQLError` data constructor is no longer being exported.

I have updated `Data.Morpheus.Types.Internal.AST.Error` to export it again. The other alternative would be to export `Msg (msg)` and `at` then you could use those to build the same `GQLError`.

Can you advise which of these you prefer (or perhaps some other solution)?

Reply:
Hi. please do not expose GQLError.  

perferded approach is to use

```hs
myError :: GQLError
myError = "my error"
```
or 
```hs
myError :: GQLError
myError = "my error" `at` (Position 1 2) 
 ```

Reply:
of course then we should use Pragma "OverloadedStrings" for that.